src/init.ml

```diff
@@ -0,0 +1,198 @@
+open! Stdune
+open! Import
+
+module Project = struct
+  module Kind = struct
+    type t =
+      | Library
+      | Executable
+
+    let to_string = function
+      | Library -> "library"
+      | Executable -> "executable"
+  end
+end
+
+module File = struct
+  type t =
+    { path: Path.t
+    ; name: string
+    ; content: string }
+
+  let make path name content =
+    {path; name; content}
+
+  let full_path {path; name; _} =
+    Path.(append path (of_string name))
+
+  let create file =
+    let path = full_path file in
+    Io.write_file ~binary:false path file.content
+end
+
+module type InitContext = sig
+  val kind : Project.Kind.t
+  val pname : string
+  val dir : Path.t
+end
+
+module type ProjectDir = sig
+  val dir : Path.t
+  val files : File.t list
+end
+
+module Proj (C : InitContext) : ProjectDir = struct
+  let dir = C.dir
+  let make_file = File.make C.dir
+
+  let dune_project =
+    let lang_stanza = Dune_project.lang_stanza () in
+    let name = Dune_project.filename in
+    let content = Printf.sprintf "\
+%s
+(name %s)
+(version 0.1)
```

Comment:
The dune-project is already generated by dune. In any case, the version number here isn't correct. I suggest that we skip (or fix) the generation of the dune-project file. Skipping seems preferable as it's already done by dune.